src/clr-angular/data/datagrid/datagrid-row.spec.ts

```diff
@@ -87,6 +87,51 @@ export default function(): void {
       });
     });
 
+    describe('Conditional selection', function() {
+      let context: TestContext<ClrDatagridRow, SelectableRow>;
+      let selectionProvider: Selection;
+
+      beforeEach(function() {
+        context = this.create(ClrDatagridRow, SelectableRow, DATAGRID_SPEC_PROVIDERS);
+        selectionProvider = TestBed.get(Selection);
+        TestBed.get(Items).all = [{ id: 1 }, { id: 2 }];
+      });
+
+      it('should not toggle when clrDgSelectable is false for type SelectionType.Single', function() {
+        selectionProvider.selectionType = SelectionType.Single;
+        context.testComponent.clrDgSelectable = false;
+        context.detectChanges();
+
+        expect(context.clarityElement.querySelector("input[type='radio']").getAttribute('disabled')).toBe('true');
+        expect(context.clarityElement.querySelector("input[type='radio']").getAttribute('aria-disabled')).toBe('true');
```

Comment:
nit: you could put the radio query in the before each to reduce a bit of duplication
`radio = context.clarityElement.querySelector("input[type='radio']"`